wordpress-mobile/WordPress-iOS
# Reader: Add reader post card cell actions

Part of #21645 
Merges into #21687

## Description

Adds:

- Tap actions to the new reader post card cell
- A "Save" action in the more menu

## Testing

To test:
- Launch Jetpack and login
- Enable the `Reader Improvements v1` feature flag if necessary
- Tap on Reader
- Tap on a **post's site title**
- 🔎 **Verify** it navigates to the site's details
- Navigate back to the reader posts
- Tap on the **Reblog** action
- 🔎 **Verify** a site picker appears
- Dismiss the site picker
- Tap on the **Comment** action
- 🔎 **Verify** it navigates to the comment section of the post
- Navigate back to the reader posts
- Tap on the **Like** action
- 🔎 **Verify** the post updates indicating the post is liked and the like count increases
- Tap on the **Liked** action
- 🔎 **Verify** the post is no longer liked and the like count decreases
- Tap on the **More (`...`)** action
- 🔎 **Verify**:
  - The actions work correctly
  - A new "Save" action appears and works correctly
- Save a post if necessary
- Navigate to your `Saved` posts
- Tap on the **More (`...`)** action for one of your saved posts
- 🔎 **Verify** a `Remove Saved Post` action appears
- Tap on `Remove Saved Post`
- 🔎 **Verify** the post is replaced by an "Undo" cell
- Tap on `Undo`
- 🔎 **Verify** the saved post reappears

**iPad**:
- Launch the app on iPad
- Navigate back to the Reader
- Tap on the **More (`...`)** action
- 🔎 **Verify** the menu appears next to the More (`...`) button

## Regression Notes
1. Potential unintended areas of impact
Removing saved posts for the old reader card

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
